src/prefect/infrastructure/process.py

```diff
@@ -168,12 +168,27 @@ async def kill(self, infrastructure_pid: str, grace_seconds: int = 30):
                     f"Unable to kill process {pid!r}: The process was not found."
                 )
 
-            await anyio.sleep(grace_seconds)
+            # Throttle how often we check if the process is still alive to keep
+            # from making too many system calls in a short period of time.
+            check_interval = max(grace_seconds / 10, 1)
+            waited_seconds = 0
+
+            while waited_seconds < grace_seconds:
```

Comment:
You can use `with anyio.move_on_after(grace_seconds)` tracking the `waited_seconds` and avoid going over